processor/samplingprocessor/tailsamplingprocessor/sampling/numeric_tag_filter.go

```diff
@@ -38,11 +43,13 @@ func NewNumericAttributeFilter(key string, minValue, maxValue int64) PolicyEvalu
 // This gives the evaluator a chance to log any message/metrics and/or update any
 // related internal state.
 func (naf *numericAttributeFilter) OnLateArrivingSpans(earlyDecision Decision, spans []*tracepb.Span) error {
+	naf.logger.Debug("Triggering action for late arriving spans in numeric-attribute filter")
```

Comment:
We generally avoid logging anything that is tied to pipeline processing. See https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md#logging

Exceptions are possible, especially since this is `Debug` which is normally disabled, so flooding is not an issue.

Just to make sure there is clear understanding of the costs. Disabled `logger.Debug` takes about 100ns. Is this acceptable cost? I assume this is printed once per batch of Spans so should be executed at a rate of hundreds to thousands of times per second. If this debug statements runs at 10,000/sec rate it will consume 0.1% of one CPU core time. This sounds as acceptable cost to me (unless I misunderstand something and this can be executed at much higher rate).

Reply:
That's right, I specifically used the `Debug` mode to avoid flooding in production. The aim was to have the logger accessible in the filters to ease development and debugging.